internal/runtime/helper.go

```diff
@@ -128,3 +130,47 @@ func constructTransactions(
 
 	return txs, nil
 }
+
+func calculateRuntimeCosts(
+	key crypto.PrivKey,
```

Comment:
You can drop half of these arguments if you just pass in a callback that signs a specific tx you give it